doc/deployment-schema.json

```diff
@@ -0,0 +1,310 @@
+                                            "required": [
+                                                "type",
+                                                "settings"
+                                            ],
+                                            "properties": {
+                                                "type": {
+                                                    "$ref": "#/definitions/moduleType"
+                                                },
+                                                "settings": {
+                                                    "$ref": "#/definitions/moduleSettings"
+                                                },
+                                                "env": {
+                                                    "$ref": "#/definitions/env"
+                                                }
+                                            }
+                                        },
+                                        "edgeHub": {
+                                            "type": "object",
+                                            "title": "The Edgehub Schema",
+                                            "required": [
+                                                "type",
+                                                "settings",
+                                                "status",
+                                                "restartPolicy"
+                                            ],
+                                            "properties": {
+                                                "type": {
+                                                    "$ref": "#/definitions/moduleType"
+                                                },
+                                                "settings": {
+                                                    "$ref": "#/definitions/moduleSettings"
+                                                },
+                                                "env": {
+                                                    "$ref": "#/definitions/env"
+                                                },
+                                                "status": {
+                                                    "$ref": "#/definitions/status"
+                                                },
+                                                "restartPolicy": {
+                                                    "$ref": "#/definitions/restartPolicy"
+                                                }
+                                            }
+                                        }
+                                    }
+                                },
+                                "modules": {
+                                    "type": "object",
+                                    "patternProperties": {
+                                        "^[a-zA-Z0-9_.-]*$": {
```

Comment:
should we use "+" instead of "*" in the regex pattern?  It should require at least 1 character.

Reply:
Is it ok to start with "-", ".", and "_"?

Reply:
fixed the pattern, it should start with a letter